### PR TITLE
Add method for remove interface

### PR DESCRIPTION
**Pull request by osmanhadzic**
Implement a method for removing the interface from the device by interface name.

Closes: #24

### CODE REVIEW

File: AstarteDeviceSDKCSharp/Device/AstarteDevice.cs [hunk 1]
```diff
@@ -251,6 +251,22 @@ public void AddInterface(string astarteInterfaceObject)
...
+        public void RemoveInterface(string interfaceName)
+        {
+            AstarteInterface? astarteInterface = GetInterface(interfaceName);
+
+            if (astarteInterface == null)
+            {
+                throw new AstarteInterfaceException("Interface " + interfaceName + " not found");
+            }
+
+            _astarteInterfaces.Remove(astarteInterface.GetInterfaceName());
```
Comment (harlem88) [inline AstarteDeviceSDKCSharp/Device/AstarteDevice.cs:267]: When an interfaces has been removed from the device,introspection should also be re-sent  to Astarte. It is the same behavior as when adding an Interface. https://github.com/astarte-platform/astarte-device-sdk-csharp/blob/a858745ea51cc74f112b0b1ca21c7b3652eb1df0/AstarteDeviceSDKCSharp/Device/AstarteDevice.cs#L255

```suggestion
            _astarteInterfaces.Remove(astarteInterface.GetInterfaceName());
            
             if (IsConnected())
            {
                _astarteTransport.SendIntrospection();
            }
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: AstarteDeviceSDKCSharp/Device/AstarteDevice.cs (hunk 1 above)
Comment (osmanhadzic) [reply AstarteDeviceSDKCSharp/Device/AstarteDevice.cs:271]: I made changes to methods to remove an added interface.

**Review by harlem88 — APPROVED** [review]

**Review by nedimtokic — APPROVED** [review]

**Review by nedimtokic — COMMENTED** [review]
@harlem88 should we add this functionality to Changelog?

**Comment by harlem88** [comment]
> @harlem88 should we add this functionality to Changelog?
I had a doubt too, because it was a feature that should have been in this changelog. `Add the capability to update the introspection dynamically.`... mmm, if two people have the same doubt, @osmanhadzic add a line about this feature in the changelog :smile:

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add the capability to update the introspection dynamically.
 - Handling sessionPresent flag from the broker.
 - Add option to ignore SSL errors in the AstarteDevice constructor.
+- Add the capability to update the introspection dynamically.
```
Comment (harlem88) [inline CHANGELOG.md:12]:
```suggestion
- Add a method to remove interface from device.
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**